community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -97,34 +101,41 @@ public boolean nodeAddLabel( KernelStatement state, long nodeId, int labelId )
         {
             PropertyConstraint constraint = constraints.next();
             int propertyKeyId = constraint.propertyKey();
-            Object value = entityReadOperations.nodeGetProperty( state, nodeId, propertyKeyId );
-            if ( value != null )
+
+            try ( Cursor<PropertyItem> properties = node.property( propertyKeyId ) )
             {
-                DefinedProperty property = Property.property( propertyKeyId, value );
-                validateNoExistingNodeWithLabelAndProperty( state, labelId, property, nodeId );
+                if ( properties.next() )
+                {
+                    DefinedProperty property = Property.property( propertyKeyId, properties.get().value() );
```

Comment:
It was already highlighted at some point, that it would be nice to avoid `DefinedProperty` creation here